### PR TITLE
Update URL when setting session.view to a saved view

### PR DESCRIPTION
**Pull request by kaixi-wang**
## What changes are proposed in this pull request?

- Pass `saved_view_slug` to server state so that the URL will update accordingly when setting session view to saved view

## How is this patch tested? If it is not, please explain why.

(Details)

### Is this a user-facing change that should be mentioned in the release notes?

-   [ ] No. You can skip the rest of this section.
-   [ ] Yes. Give a description of this change to be included in the release
        notes for FiftyOne users.

(Details in 1-2 sentences. You can just refer to another PR with a description
if this PR is part of a larger change.)

### What areas of FiftyOne does this PR affect?

-   [ ] App: FiftyOne application changes
-   [ ] Build: Build and test infrastructure changes
-   [ ] Core: Core `fiftyone` Python library changes
-   [ ] Documentation: FiftyOne documentation changes
-   [ ] Other

### CODE REVIEW

**Review by imanjra — APPROVED** [review]
lgtm 👍

File: app/packages/core/src/components/Sidebar/ViewSelection/index.tsx [hunk 1]
```diff
@@ -41,7 +41,8 @@ export default function ViewSelection() {
   const datasetName = useRecoilValue(fos.datasetName);
   const canEditSavedViews = useRecoilValue<boolean>(fos.canEditSavedViews);
   const setIsOpen = useSetRecoilState<boolean>(viewDialogOpen);
-  const savedViewParam = useRecoilValue(fos.viewName);
+  const urlParams = new URLSearchParams(window.location.search);
+  const savedViewParam = useRecoilValue(fos.viewName) || urlParams.get("view");
```
Comment (benjaminpkane) [inline app/packages/core/src/components/Sidebar/ViewSelection/index.tsx:45]: Hmm, this shouldn't be necessary and is defaulting to the slug and not the name. Let me try to push a tweak

**Review by benjaminpkane — CHANGES_REQUESTED** [review]

File: app/packages/core/src/components/Sidebar/ViewSelection/index.tsx (hunk 1 above)
Comment (manivoxel51) [reply app/packages/core/src/components/Sidebar/ViewSelection/index.tsx:45]: yes! will remove it - was experimenting

Comment (kaixi-wang) [reply app/packages/core/src/components/Sidebar/ViewSelection/index.tsx:45]: @benjaminpkane did you get a chance to check this?

**Review by benjaminpkane — APPROVED** [review]
LGTM. Simplifying `StateDescription` to hold just the `view` got things working

File: app/packages/core/src/components/Sidebar/ViewSelection/index.tsx (hunk 1 above)
Comment (benjaminpkane) [reply app/packages/core/src/components/Sidebar/ViewSelection/index.tsx]: Things are working now

**Pull request merged**